### PR TITLE
[DESK-1142] SSH launch

### PR DESCRIPTION
**Pull request by mauricio0000**
### Description

[DESK-1142] SSH launch

### Ticket

https://remoteit.atlassian.net/browse/DESK-1142

[DESK-1142]: https://remoteit.atlassian.net/browse/DESK-1142

### CODE REVIEW

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
I saw this dialog on my mac:
Screen Shot 2021-07-12 at 3 50 21 PM

It should only appear on windows when putty isn't available.
Also it should remember the state the user chooses in memory so they only have to do it once every time they start the app.

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
When I tried launching an ssh connection with a missing token [username] it did not prompt me for the username.

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
@mauricio0000 Mac testing was good. But didn't pass my Windows testing. Got stuck on this screen:

Screen Shot 2021-07-23 at 9 30 52 AM

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Let's re-review - still having issues launching in Windows:

Error:
Screen Shot 2021-08-03 at 11 34 46 AM

And showing Putty when launching RDP:
Screen Shot 2021-08-03 at 11 36 10 AM

I think we should fix this the right way and integrate with the `frontend/src/shared/applications.ts` class. Let's discuss. 

Thanks!

File: frontend/src/buttons/LaunchButton/LaunchButton.tsx [hunk 1]
```diff
@@ -34,103 +31,90 @@ export const LaunchButton: React.FC<Props> = ({
...
-        setDownloadLink('https://www.realvnc.com/en/connect/download/viewer/windows/')
-        setOpenApp(true)
-        ui.set({ requireInstall: 'none' })
-        break
+    if (openLaunchApplication && !loading) {
+      launchApplication()
+      setOpenLaunchApplication(false)
     }
-  }, [requireInstall, launch, app])
+  }, [loading])
```
Comment (javierdac) [inline frontend/src/buttons/LaunchButton/LaunchButton.tsx:55]: Is it not necessary to add openLaunchApplication in here?

File: backend/src/sharedCopy/applications.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export class Application {
   launchIcon: string = 'launch'
   commandIcon: string = 'terminal'
   publicTemplate: string = '[address]'
+  defaultTemplateCmd: string = ''
+  checkApplicationCmd: string = ''
```
[the remark this answers is not in the document]
Comment (mauricio0000) [reply backend/src/sharedCopy/applications.ts:19]: by default it is empty

File: frontend/src/components/DialogApp/DialogApp.tsx [hunk 3]
```diff
@@ -1,32 +1,65 @@
...
+          </DialogContentText>
+        </DialogContent>
         <DialogActions>
-          <Button onClick={closeAll} color="primary">
-            Cancel
-          </Button>
-          <Button onClick={getApp} variant="contained" color="primary">
-            Download {App}
-          </Button>
+          <Button onClick={getApp} style={{ borderRadius: 3, left: '-120px' }} variant="contained" color="primary"   >{buttonText}</Button>
```
Comment (javierdac) [inline frontend/src/components/DialogApp/DialogApp.tsx:58]: we should add a stylesheet on this

File: frontend/src/services/Browser.ts [hunk 4]
```diff
@@ -60,3 +60,30 @@ export function launchVNC(typeID?: number) {
 export function launchRemoteDesktop(typeID?: number) {
   return typeID === 5 && isWindows()
 }
+
+export function getApplicationObj(typeID?: number, username?: string) {
+  if (launchPutty(typeID)) {
```
Comment (javierdac) [inline frontend/src/services/Browser.ts:65]: should not be better to use switch in here?

**Review by javierdac — CHANGES_REQUESTED** [review]

**Pull request merged**